File: examples/official-x402-client.py
# Register safe-pay on the OFFICIAL python x402 client's pre-pay hook.
#
# The client calls `on_before_payment_creation(req)` before signing; `req.pay_to`
# is the recipient. Return an abort decision to block (shape per the client's
# hook contract; abort/continue below is illustrative).
#
# Reference snippet — `paid_post(url, json_body) -> dict` is your x402-capable POST.
from anchor_safe_pay import screen_allows


def on_before_payment_creation(req):
    ok, verdict = screen_allows(req.pay_to, fetch=paid_post)
    if not ok:
        return {"abort": True, "reason": f"recipient {verdict['recommendation']}"}
    return None  # continue

File: examples/official-x402-client.mjs
// Register safe-pay on the OFFICIAL @x402 client's pre-pay hook.
//
// `beforePaymentCreation` fires with the selected payment requirements
// (including `payTo`) BEFORE anything is signed; return {abort, reason} to
// block. Works with wrapFetchWithPayment / wrapAxios unchanged.
//
// Reference snippet — assumes you've built `x402Client` and have a paid fetch.
import { screenAllows } from "anchor-x402-safe-pay";

// `paidFetch` is your x402-capable fetch used ONLY to settle the $0.02 screen
// call — keep it separate from the client being guarded so screening a payment
// never recurses into itself.
x402Client.onBeforePaymentCreation(async (req) => {
  const { ok, verdict } = await screenAllows(req.payTo, { fetchImpl: paidFetch });
  return ok
    ? undefined // continue
    : { abort: true, reason: `recipient ${verdict.recommendation} (risk ${verdict.risk_score})` };
});
